astropy/tests/command.py

```diff
@@ -29,6 +29,22 @@ def to_str_or_none(x):
     return [tuple(to_str_or_none(x) for x in y) for y in options]
 
 
+class FixRemoteDataOption(type):
+    """
+    This metaclass is used to catch cases where the user is running the tests
+    with --remote-data. We've now changed the --remote-data option so that it
+    takes arguments, but we still want --remote-data to work as before and to
+    enable all remote tests. With this metaclass, we can modify sys.argv
+    before distutils/setuptools try to parse the command-line options.
+    """
+    def __init__(cls, name, bases, dct):
+        for i in range(len(sys.argv)):
```

Comment:
The `range(len(...))` is something of a do-not in my opinion, I think a more elegant way would be:

```
try:
    idx = sys.argv.index('--remote-data')
except ValueError:
    pass
else:
    sys.argv[idx] = '--remote-data=any'
```

but it just does exactly the same, so keep whichever you like more.